aws_topology/stackstate_checks/aws_topology/resources/sqs.py

```diff
@@ -1,82 +1,111 @@
-from .utils import make_valid_data, with_dimensions, create_arn as arn
+from .utils import (
+    make_valid_data,
+    with_dimensions,
+    create_arn as arn,
+    client_array_operation,
+    set_required_access_v2,
+    transformation,
+)
 from .registry import RegisteredResourceCollector
+from collections import namedtuple
+from schematics import Model
+from schematics.types import StringType
 import re
 
 
 def create_arn(region=None, account_id=None, resource_id=None, **kwargs):
     match_string = r"^https:\/\/sqs.[a-z]{2}-([a-z]*-){1,2}\d\.amazonaws.com\/\d{12}\/.+$"
     if re.match(match_string, resource_id):
         return arn(
-            resource='sqs',
-            region=resource_id.split('.', 2)[1],
-            account_id=resource_id.rsplit('/', 2)[-2],
-            resource_id=resource_id.rsplit('/', 1)[-1],
+            resource="sqs",
+            region=resource_id.split(".", 2)[1],
+            account_id=resource_id.rsplit("/", 2)[-2],
+            resource_id=resource_id.rsplit("/", 1)[-1],
         )
     else:
         raise ValueError("SQS URL {} does not match expected regular expression {}".format(resource_id, match_string))
 
 
+QueueData = namedtuple("QueueData", ["queue_url", "queue", "tags"])
+
+
+class Queue(Model):
+    QueueArn = StringType(required=True)
+
+
 class SqsCollector(RegisteredResourceCollector):
     API = "sqs"
     API_TYPE = "regional"
     COMPONENT_TYPE = "aws.sqs"
-    CLOUDFORMATION_TYPE = 'AWS::SQS::Queue'
+    CLOUDFORMATION_TYPE = "AWS::SQS::Queue"
+
+    @set_required_access_v2("sqs:ListQueueTags")
+    def collect_tags(self, queue_url):
+        return self.client.list_queue_tags(QueueUrl=queue_url).get("Tags", [])
+
+    @set_required_access_v2("sqs:GetQueueAttributes")
+    def collect_queue_description(self, queue_url):
+        return self.client.get_queue_attributes(QueueUrl=queue_url, AttributeNames=["All"]).get("Attributes", {})
+
+    def construct_queue_description(self, queue_url):
+        return {
+            "QueueArn": self.agent.create_arn("AWS::SQS::Queue", self.location_info, resource_id=queue_url)
```

Comment:
Adding the same question here, just resolve them if it's not relevant. 

Can we use the queue ARN function?

Reply:
Same answer, it uses that function, but in a roundabout way for consistency when doing cross-api create_arn calls.